### PR TITLE
Generic name change - To maintain a generic name instead of "image"

### PR DESCRIPTION
**Pull request by swetha097**
This PR changes the names of image_info to sample_info to maintain a generic name for all use-cases (audio, video, image)

### CODE REVIEW

**Comment by rrawther** [comment]
@LakshmiKumar23 : can you please review this PR?

File: rocAL/source/pipeline/master_graph.cpp [hunk 1]
```diff
@@ -970,7 +970,7 @@ void MasterGraph::output_routine() {
             _sequence_start_framenum_vec.insert(_sequence_start_framenum_vec.begin(), _loader_module->get_sequence_start_frame_number());
             _sequence_frame_timestamps_vec.insert(_sequence_frame_timestamps_vec.begin(), _loader_module->get_sequence_frame_timestamps());
 #endif
-            _ring_buffer.set_meta_data(full_batch_image_names, output_meta_data);
+            _ring_buffer.set_meta_data(full_batch_sample_names, output_meta_data);
             _ring_buffer.push();  // Image data and metadata is now stored in output the ring_buffer, increases it's level by 1
```
Comment (LakshmiKumar23) [inline rocAL/source/pipeline/master_graph.cpp:974]: @swetha097 comment still refers to "image"

File: rocAL/include/loaders/circular_buffer.h [hunk 2]
```diff
@@ -54,9 +54,9 @@ class CircularBuffer {
     void unblock_writer();  // Unblocks the thread currently waiting on get_write_buffer
     void push();            // The latest write goes through, effectively adds one element to the buffer
     void pop();             // The oldest write will be erased and overwritten in upcoming writes
-    void set_image_info(const decoded_image_info& info) { _last_image_info = info; }
+    void set_sample_info(const decoded_sample_info& info) { _last_sample_info = info; }
     void set_crop_image_info(const crop_image_info& info) { _last_crop_image_info = info; }
```
Comment (LakshmiKumar23) [inline rocAL/include/loaders/circular_buffer.h:58]: @swetha097 Are we not changing this also for uniformity?

File: rocAL/include/loaders/circular_buffer.h [hunk 3]
```diff
@@ -54,9 +54,9 @@ class CircularBuffer {
...
-    decoded_image_info& get_image_info();
+    decoded_sample_info& get_sample_info();
     crop_image_info& get_cropped_image_info();
```
Comment (LakshmiKumar23) [inline rocAL/include/loaders/circular_buffer.h:60]: @swetha097 same as above "crop_sample_info"? Seems like a few places

**Review by rrawther — COMMENTED** [review]
decode_sample_info is too confusing which can be interpreted for audio. Why are we changing the existing naming?

File: rocAL/include/loaders/circular_buffer.h (hunk 3 above)
Comment (swetha097) [reply rocAL/include/loaders/circular_buffer.h:60]: The crop_image_info is only specific to images - for random bbox crops. Hence did not make it generic

File: rocAL/source/pipeline/master_graph.cpp (hunk 1 above)
Comment (swetha097) [reply rocAL/source/pipeline/master_graph.cpp]: Changed

**Comment by shobana-mcw** [comment]
Hi @rrawther,

In our last call we discussed to rename image to sample which will be a common reference for image / audio / video. We are using this API to get info for other media too. 

Can we replace sample with media / data? Please share your thoughts.

File: rocAL/include/meta_data/bounding_box_graph.h [hunk 4]
```diff
@@ -31,8 +31,8 @@ typedef  struct { float xc; float yc; float w; float h; } BoundingBoxCord_xcycwh
 class BoundingBoxGraph : public MetaDataGraph {
    public:
     void process(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data) override;
-    void update_meta_data(pMetaDataBatch meta_data, decoded_image_info decode_image_info) override;
-    void update_random_bbox_meta_data(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data, decoded_image_info decoded_image_info, crop_image_info crop_image_info) override;
+    void update_meta_data(pMetaDataBatch meta_data, decoded_sample_info decode_image_info) override;
```
Comment (rrawther) [inline rocAL/include/meta_data/bounding_box_graph.h:34]: Is the meta_data always corresponds to image as well? Also I am noticing the struct name and variable name is same here. Google coding style required camelcase for struct and snake_case for variable names

File: rocAL/include/loaders/circular_buffer.h [hunk 5]
```diff
@@ -31,8 +31,8 @@ THE SOFTWARE.
 #include "commons.h"
 #include "device_manager.h"
 #include "device_manager_hip.h"
-struct decoded_image_info {
-    std::vector<std::string> _image_names;
+struct decoded_sample_info {
```
Comment (rrawther) [inline rocAL/include/loaders/circular_buffer.h:34]: please change the struct name to camel case. Instead of naming _sample_names, you can have a union which can hold image_names, video_file_names or audio_sample_name. That way individual reader can use appropriate names

Comment (rrawther) [inline rocAL/include/loaders/circular_buffer.h:34]: "data" is more appropriate name than sample

**Review by rrawther — CHANGES_REQUESTED** [review]
Added few comments. Please change accordingly

File: rocAL/include/loaders/circular_buffer.h (hunk 5 above)
Comment (swetha097) [reply rocAL/include/loaders/circular_buffer.h]: Changed to DecodedDataInfo

Comment (swetha097) [reply rocAL/include/loaders/circular_buffer.h]: Struct name changed to camelCase. Working on the union part.

File: rocAL/include/loaders/loader_module.h [hunk 6]
```diff
@@ -53,8 +53,8 @@ class LoaderModule {
     virtual Timing timing() = 0;                    // Returns timing info
     virtual std::vector<std::string> get_id() = 0;  // returns the id of the last batch of images/frames loaded
     virtual void start_loading() = 0;               // starts internal loading thread
-    virtual decoded_image_info get_decode_image_info() = 0;
-    virtual crop_image_info get_crop_image_info() = 0;
+    virtual DecodedDataInfo get_decode_data_info() = 0;
+    virtual crop_image_info get_crop_image_info() { return {}; }
```
Comment (rrawther) [inline rocAL/include/loaders/loader_module.h:57]: why is this changed?

File: rocAL/source/loaders/circular_buffer.cpp [hunk 7]
```diff
@@ -40,8 +40,8 @@ void CircularBuffer::reset() {
     _write_ptr = 0;
     _read_ptr = 0;
     _level = 0;
-    while (!_circ_image_info.empty())
-        _circ_image_info.pop();
+    while (!_circ_data_info.empty())
```
Comment (rrawther) [inline rocAL/source/loaders/circular_buffer.cpp:43]: change it to _circ_buff_data_info

**Review by rrawther — CHANGES_REQUESTED** [review]
added some comments

File: rocAL/include/loaders/circular_buffer.h [hunk 8]
```diff
@@ -73,8 +73,8 @@ class CircularBuffer {
     bool full();
     bool empty();
     size_t _buff_depth;
-    decoded_image_info _last_image_info;
-    std::queue<decoded_image_info> _circ_image_info;    //!< Stores the loaded images names, decoded_width and decoded_height(data is stored in the _circ_buff)
+    DecodedDataInfo _last_data_info;
+    std::queue<DecodedDataInfo> _circ_data_info;    //!< Stores the loaded data names, decoded_width and decoded_height(data is stored in the _circ_buff)
     crop_image_info _last_crop_image_info;              // for Random BBox crop coordinates
     std::queue<crop_image_info> _circ_crop_image_info;  //!< Stores the crop coordinates of the images for random bbox crop (data is stored in the _circ_buff)
```
Comment (rrawther) [inline rocAL/include/loaders/circular_buffer.h:79]: what is actual_host_buffers in line #95 and why is it a vector of vector of chars. If it is just buffer pointers, please declare as uint8_t * pointers.

File: rocAL/include/loaders/video/video_loader.h [hunk 9]
```diff
@@ -74,8 +74,8 @@ class VideoLoader : public LoaderModule {
     size_t _sequence_length;
     std::thread _load_thread;
     RocalMemType _mem_type;
-    decoded_image_info _decoded_img_info;
-    decoded_image_info _output_decoded_img_info;
+    DecodedDataInfo _decoded_video_info;
```
Comment (rrawther) [inline rocAL/include/loaders/video/video_loader.h:77]: if decoded_data_info instances are required for all loaders, it is better to define in one place in the base class

File: rocAL/include/loaders/circular_buffer.h [hunk 10]
```diff
@@ -31,8 +31,8 @@ THE SOFTWARE.
 #include "commons.h"
 #include "device_manager.h"
 #include "device_manager_hip.h"
-struct decoded_image_info {
-    std::vector<std::string> _image_names;
+struct DecodedDataInfo {
+    std::vector<std::string> _data_names;
```
Comment (rrawther) [inline rocAL/include/loaders/circular_buffer.h:35]: please rename crop_image_info to CropImageInfo

File: rocAL/include/loaders/image/image_loader.h [hunk 11]
```diff
@@ -74,9 +74,9 @@ class ImageLoader : public LoaderModule {
...
     std::thread _load_thread;
     RocalMemType _mem_type;
-    decoded_image_info _decoded_img_info;
+    DecodedDataInfo _decoded_img_info;
     crop_image_info _crop_image_info;
-    decoded_image_info _output_decoded_img_info;
+    DecodedDataInfo _output_decoded_img_info;
     crop_image_info _output_cropped_img_info;
     CircularBuffer _circ_buff;
     TimingDBG _swap_handle_time;
```
Comment (LakshmiKumar23) [inline rocAL/include/loaders/image/image_loader.h:82]: Change to TimingDbg

File: rocAL/include/loaders/image/image_loader.h [hunk 12]
```diff
@@ -74,9 +74,9 @@ class ImageLoader : public LoaderModule {
     size_t _batch_size;
     std::thread _load_thread;
     RocalMemType _mem_type;
-    decoded_image_info _decoded_img_info;
+    DecodedDataInfo _decoded_img_info;
     crop_image_info _crop_image_info;
```
Comment (LakshmiKumar23) [inline rocAL/include/loaders/image/image_loader.h:78]: change to CropImageInfo

File: rocAL/include/loaders/circular_buffer.h [hunk 13]
```diff
@@ -73,8 +73,8 @@ class CircularBuffer {
     bool full();
     bool empty();
     size_t _buff_depth;
-    decoded_image_info _last_image_info;
-    std::queue<decoded_image_info> _circ_image_info;    //!< Stores the loaded images names, decoded_width and decoded_height(data is stored in the _circ_buff)
+    DecodedDataInfo _last_data_info;
+    std::queue<DecodedDataInfo> _circ_data_info;    //!< Stores the loaded data names, decoded_width and decoded_height(data is stored in the _circ_buff)
     crop_image_info _last_crop_image_info;              // for Random BBox crop coordinates
     std::queue<crop_image_info> _circ_crop_image_info;  //!< Stores the crop coordinates of the images for random bbox crop (data is stored in the _circ_buff)
     std::mutex _names_buff_lock;
```
Comment (LakshmiKumar23) [inline rocAL/include/loaders/circular_buffer.h:80]: Commented lines 81-84, which variable is that for? It's hanging without reference

File: rocAL/source/pipeline/master_graph.cpp [hunk 14]
```diff
@@ -903,16 +903,16 @@ void MasterGraph::output_routine() {
...
-            auto full_batch_image_names = _loader_module->get_id();
-            auto decode_image_info = _loader_module->get_decode_image_info();
+            auto full_batch_data_names = _loader_module->get_id();
+            auto decode_data_info = _loader_module->get_decode_data_info();
             auto crop_image_info = _loader_module->get_crop_image_info();
 
-            if (full_batch_image_names.size() != _user_batch_size)
-                WRN("Internal problem: names count " + TOSTR(full_batch_image_names.size()))
+            if (full_batch_data_names.size() != _user_batch_size)
+                WRN("Internal problem: names count " + TOSTR(full_batch_data_names.size()))
```
Comment (LakshmiKumar23) [inline rocAL/source/pipeline/master_graph.cpp:911]: what does "internal problem" mean? Just say "Master Graph - batch count doesn't match label count" or something

**Comment by LakshmiKumar23** [comment]
@swetha097 can you address the review comments? We need to merge this before we can review the audio PRs

File: rocAL/include/loaders/loader_module.h (hunk 6 above)
Comment (swetha097) [reply rocAL/include/loaders/loader_module.h:57]: Changed from pure virtual to virtual since its not used in the Audio. This API is only related to image.

File: rocAL/include/loaders/circular_buffer.h (hunk 8 above)
Comment (swetha097) [reply rocAL/include/loaders/circular_buffer.h]: Removed this variable as its unused

File: rocAL/include/loaders/circular_buffer.h (hunk 10 above)
Comment (swetha097) [reply rocAL/include/loaders/circular_buffer.h:35]: Renamed it

File: rocAL/include/loaders/video/video_loader.h (hunk 9 above)
Comment (swetha097) [reply rocAL/include/loaders/video/video_loader.h]: Moved to the base class

File: rocAL/source/pipeline/master_graph.cpp (hunk 14 above)
Comment (swetha097) [reply rocAL/source/pipeline/master_graph.cpp]: Changed

File: rocAL/include/loaders/circular_buffer.h (hunk 13 above)
Comment (swetha097) [reply rocAL/include/loaders/circular_buffer.h:80]: Removed the comment

File: rocAL/source/loaders/circular_buffer.cpp (hunk 7 above)
Comment (swetha097) [reply rocAL/source/loaders/circular_buffer.cpp]: Addressed

File: rocAL/include/meta_data/bounding_box_graph.h (hunk 4 above)
Comment (swetha097) [reply rocAL/include/meta_data/bounding_box_graph.h]: bounding_box_graph file is meant to handle images meta-data only.

**Comment by swetha097** [comment]
@LakshmiKumar23 Addressed the comments.

File: rocAL/include/loaders/video/video_loader.h [hunk 15]
```diff
@@ -74,10 +74,8 @@ class VideoLoader : public LoaderModule {
...
     std::thread _load_thread;
     RocalMemType _mem_type;
-    decoded_image_info _decoded_img_info;
-    decoded_image_info _output_decoded_img_info;
     CircularBuffer _circ_buff;
-    TimingDBG _swap_handle_time;
+    TimingDbg _swap_handle_time;
     bool _is_initialized;
     bool _stopped = false;
     bool _loop;                         //<! If true the reader will wrap around at the end of the media (files/images/...) and wouldn't stop
```
Comment (LakshmiKumar23) [inline rocAL/include/loaders/video/video_loader.h:81]: @swetha097 What files/images are you referring to?

File: rocAL/include/meta_data/meta_data_graph.h [hunk 16]
```diff
@@ -41,8 +41,8 @@ class MetaDataGraph {
    public:
     virtual ~MetaDataGraph() = default;
     virtual void process(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data) = 0;
-    virtual void update_meta_data(pMetaDataBatch meta_data, decoded_image_info decoded_image_info) = 0;
-    virtual void update_random_bbox_meta_data(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data, decoded_image_info decoded_image_info, crop_image_info crop_image_info) = 0;
+    virtual void update_meta_data(pMetaDataBatch meta_data, DecodedDataInfo decoded_image_info) = 0;
```
[the remark this answers is not in the document]
Comment (swetha097) [reply rocAL/include/meta_data/meta_data_graph.h:44]: Changed to decoded_data_info.

File: rocAL/include/loaders/video/video_loader.h (hunk 15 above)
Comment (sbavasab) [reply rocAL/include/loaders/video/video_loader.h:81]: Files/ images here refers to video frames

**Review by LakshmiKumar23 — APPROVED** [review]
@kiritigowda can be merged. All review comments have been addressed

**Comment by kiritigowda** [comment]
@swetha097 @LakshmiKumar23 -- failing CI tests - http://math-ci.amd.com/blue/organizations/jenkins/main%2Fprecheckin%2FrocAL/detail/PR-114/2/pipeline/227

**Comment by kiritigowda** [comment]
@swetha097  @LakshmiKumar23 

```
[ 68%] Building CXX object rocAL/CMakeFiles/rocal.dir/source/meta_data/meta_node_resize_mirror_normalize.cpp.o
../../../rocAL/source/meta_data/bounding_box_graph.cpp: In member function 'virtual void BoundingBoxGraph::update_random_bbox_meta_data(pMetaDataBatch, pMetaDataBatch, DecodedDataInfo, CropImageInfo)':
../../../rocAL/source/meta_data/bounding_box_graph.cpp:76:36: error: expected primary-expression before '.' token
   76 |     auto crop_cords = CropImageInfo._crop_image_coords;
      |                                    ^
make[2]: *** [rocAL/CMakeFiles/rocal.dir/build.make:1060: rocAL/CMakeFiles/rocal.dir/source/meta_data/bounding_box_graph.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:909: rocAL/CMakeFiles/rocal.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

```

**Comment by swetha097** [comment]
@kiritigowda Fixed it.

**Review by rrawther — APPROVED** [review]

**Comment by rrawther** [comment]
@kiritigowda : this is ready to be merged. Is it still failing CI?

**Comment by LakshmiKumar23** [comment]
@kiritigowda CI passed on this
